modules/repository-s3/src/test/java/org/elasticsearch/repositories/s3/RepositoryCredentialsTests.java

```diff
@@ -267,7 +267,7 @@ protected S3Repository createRepository(
         ) {
             return new S3Repository(metadata, registry, getService(), clusterService, bigArrays, recoverySettings, s3RepositoriesMetrics) {
                 @Override
-                protected void assertSnapshotOrGenericThread() {
+                protected void assertSnapshotOrStatelessPermittedThreadPool() {
                     // eliminate thread name check as we create repo manually on test/main threads
```

Comment:
Are you sure these overrides are needed any more? The check in `S3Repository` permits things to happen on the main test thread these days.

Reply:
It would appear that deleting the override from `BlobStoreRepositoryRestoreTests` works locally.

I'm not inclined to make code changes in this patch, but I'll take a look 👍